File: day_03/part_1/solution.py
import re

with open("../input.txt", 'r') as file:
    memory = file.read()

pattern = re.compile(r'mul\((\d+),(\d+)\)')
matches = pattern.findall(memory)

total_sum = 0
for match in matches:
    x = int(match[0])
    y = int(match[1])
    total_sum += x * y

print("Answer:", total_sum)

File: day_03/part_1/solution.js
const fs = require('fs');

const memory = fs.readFileSync('../input.txt', 'utf8');

const pattern = /mul\((\d+),(\d+)\)/g;
let matches;
let totalSum = 0;

while ((matches = pattern.exec(memory)) !== null) {
    const x = parseInt(matches[1], 10);
    const y = parseInt(matches[2], 10);
    totalSum += x * y;
}

console.log("Answer:", totalSum);